plugins/funind/glob_termops.mli

```diff
@@ -122,3 +122,10 @@ val zeta_normalize : Glob_term.glob_constr -> Glob_term.glob_constr
 
 
 val expand_as : glob_constr -> glob_constr
+
+
+(* [resolve_and_replace_implicits ?expected_type env sigma rt] solves implicits of [rt] w.r.t. [env] and [sigma] and then replace them by their solution 
+ *)
+val resolve_and_replace_implicits :
```

Comment:
@forestjulien: The pattern-matching algorithm is actually polymorphic on the type of the right-hand side of the clauses. It can take a pattern-matching problem in `constr` and for that we just have to pass it the function which will interpret the right-hand sides.

However, I have no idea how easily `Function` can take benefit of this.